frontend/src/shared/components/Toast/store/toastActions.ts

```diff
@@ -0,0 +1,103 @@
+import { DEFAULT_DURATION_MS, DEFAULT_TYPE } from "../constants/toast.constants";
+import type { ToastData, ToastOptions, ToastsState } from "../types/toast.type";
+import { toastStore } from "./toastStore";
+
+const getDistributedToasts = (state: ToastsState) => {
+  const { toasts, defaultPosition, limit } = state;
+  const queue: ToastData[] = [];
+  const activeToasts: ToastData[] = [];
+  const count: Record<string, number> = {};
+
+  toasts.forEach((toast) => {
+    const position = toast.position || defaultPosition;
+    count[position] = count[position] || 0;
+    count[position] += 1;
+
+    if (count[position] <= limit) {
+      activeToasts.push(toast);
+    } else {
+      queue.push(toast);
+    }
+  });
+
+  return { filteredToasts: activeToasts, queue };
+};
+
+export const showToast = (options: ToastOptions) => {
+  const { message, type = DEFAULT_TYPE, position, duration = DEFAULT_DURATION_MS } = options;
+
+  const id = crypto.randomUUID();
+  const toast: ToastData = {
+    id,
+    message,
+    type,
+    position,
+    duration,
+    createdAt: Date.now(),
+  };
+
+  const currentState = toastStore.getState();
+
+  const isDuplicate = currentState.toasts.some(({ message: toastMessage, type: toastType }) => toastMessage === message && toastType === type);
```

Comment:
```suggestion
  const duplicated = currentState.toasts.some((current) => current.message === message && current.type === type);
```
저는 요새 이런 경우에는 구조 분해 할당을 사용하지 않는 것 같네요!